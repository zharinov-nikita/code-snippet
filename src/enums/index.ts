export * from './template-name.enum'