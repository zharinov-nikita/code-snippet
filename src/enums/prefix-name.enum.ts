import type { TypeCase } from '../types'

export const enumPrefixName: TypeCase = {
  camelCase: 'prefixName',
  pascalCase: 'PrefixName',
  lowerSnakeCase: 'prefix_name',
  upperSnakeCase: 'PREFIX_NAME',
  lowerKebabCase: 'prefix-name',
  upperKebabCase: 'PREFIX-NAME',
}
