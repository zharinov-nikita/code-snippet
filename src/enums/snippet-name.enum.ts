export enum enumSnippetName {
  camelCase = 'snippetName',
  pascalCase = 'SnippetName',
  lowerSnakeCase = 'snippet_name',
  upperSnakeCase = 'SNIPPET_NAME',
  lowerKebabCase = 'snippet-name',
  upperKebabCase = 'SNIPPET-NAME',
}
