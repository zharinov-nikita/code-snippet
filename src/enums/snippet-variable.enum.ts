import type { TypeCase } from '../types'

export const enumSnippetVariable: TypeCase = {
    camelCase: 'snippetVariable',
    pascalCase: 'SnippetVariable',
    lowerSnakeCase: 'snippet_variable',
    upperSnakeCase: 'SNIPPET_VARIABLE',
    lowerKebabCase: 'snippet-variable',
    upperKebabCase: 'SNIPPET-VARIABLE',
}
