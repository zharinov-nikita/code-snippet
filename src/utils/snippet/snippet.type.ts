import type { TypeGeneratePath } from '../../types'

export interface TypeFormatOptions {
  pathToSnippet: string
  name: string
}

export interface TypeCreateOptions extends TypeGeneratePath {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  formattedSnippet: any
  isFlat: boolean
}
