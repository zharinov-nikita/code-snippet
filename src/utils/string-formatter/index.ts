export * from './string-formatter.util'
