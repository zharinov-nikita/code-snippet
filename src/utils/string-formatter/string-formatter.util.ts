export class UtilStringFormatter {
  toCamelCase(stringInKebabCase: string): string {
    const array = stringInKebabCase.split('-')
    const firstWord = array[0]
    const otherWord = array.map((item, index) => {
      if (index > 0) {
        return `${item[0].toUpperCase()}${item.substring(1)}`
      }
    })

    return `${firstWord}${otherWord.join('')}`
  }

  toUpperSnakeCase(stringInKebabCase: string): string {
    const array = stringInKebabCase.split('-')
    const string = array.map((item) => item.toUpperCase())
    return string.join('_')
  }

  toLowerSnakeCase(stringInKebabCase: string): string {
    const array = stringInKebabCase.split('-')
    const string = array.map((item) => item.toLowerCase())
    return string.join('_')
  }

  toUpperKebabCase(stringInKebabCase: string): string {
    return stringInKebabCase.toUpperCase()
  }

  toLowerKebabCase(stringInKebabCase: string): string {
    return stringInKebabCase.toLowerCase()
  }

  toPascalCase(stringInKebabCase: string): string {
    const array = stringInKebabCase.split('-')
    const firstWord = `${array[0][0].toUpperCase()}${array[0].substring(1)}`
    const otherWord = array.map((item, index) => {
      if (index > 0) {
        return `${item[0].toUpperCase()}${item.substring(1)}`
      }
    })

    return `${firstWord}${otherWord.join('')}`
  }
}
