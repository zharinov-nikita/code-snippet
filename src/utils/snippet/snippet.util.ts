import { enumSnippetName } from '../../enums'
import { UtilStringFormatter } from '../string-formatter'

export class UtilSnippet {
  private stringFormatter: UtilStringFormatter

  constructor() {
    this.stringFormatter = new UtilStringFormatter()
  }

  public replace(snippet: string, name: string): string {
    return snippet
      .replaceAll(enumSnippetName.camelCase, `${this.stringFormatter.toCamelCase(name)}`)
      .replaceAll(enumSnippetName.pascalCase, `${this.stringFormatter.toPascalCase(name)}`)
      .replaceAll(enumSnippetName.lowerSnakeCase, `${this.stringFormatter.toLowerSnakeCase(name)}`)
      .replaceAll(enumSnippetName.upperSnakeCase, `${this.stringFormatter.toUpperSnakeCase(name)}`)
      .replaceAll(enumSnippetName.lowerKebabCase, `${this.stringFormatter.toLowerKebabCase(name)}`)
      .replaceAll(enumSnippetName.upperKebabCase, `${this.stringFormatter.toUpperKebabCase(name)}`)
  }
}
