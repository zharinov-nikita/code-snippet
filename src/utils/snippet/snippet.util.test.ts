import { describe, expect, test } from '@jest/globals'
import { enumSnippetName } from '../../enums'
import { UtilStringFormatter } from '../string-formatter'
import { UtilSnippet } from './snippet.util'

describe('util/snippet', () => {
  const utilSnippet = new UtilSnippet()
  const utilStringFormatter = new UtilStringFormatter()
  const name = 'hello-world'

  const snippet = `
    [${enumSnippetName.camelCase}]
    [${enumSnippetName.lowerKebabCase}]
    [${enumSnippetName.lowerSnakeCase}]
    [${enumSnippetName.pascalCase}]
    [${enumSnippetName.upperKebabCase}]
    [${enumSnippetName.upperSnakeCase}]
    `
  const formattedSnippet = `
    [${utilStringFormatter.toCamelCase(name)}]
    [${utilStringFormatter.toLowerKebabCase(name)}]
    [${utilStringFormatter.toLowerSnakeCase(name)}]
    [${utilStringFormatter.toPascalCase(name)}]
    [${utilStringFormatter.toUpperKebabCase(name)}]
    [${utilStringFormatter.toUpperSnakeCase(name)}]
    `

  test('toCase', () => {
    expect(utilSnippet.replace(snippet, name)).toBe(formattedSnippet)
  })

  test('snapshot', () => {
    expect(utilSnippet).toMatchSnapshot('util')
    expect(utilSnippet.replace(snippet, name)).toMatchSnapshot('replace')
  })
})
