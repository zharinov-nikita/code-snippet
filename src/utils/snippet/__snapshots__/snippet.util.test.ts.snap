// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util/snippet snapshot: replace 1`] = `
"
    [helloWorld]
    [hello-world]
    [hello_world]
    [HelloWorld]
    [HELLO-WORLD]
    [HELLO_WORLD]
    "
`;

exports[`util/snippet snapshot: util 1`] = `
UtilSnippet {
  "stringFormatter": UtilStringFormatter {},
}
`;
